pkg/controller/pxc/upgrade.go

```diff
@@ -208,7 +204,7 @@ func (r *ReconcilePerconaXtraDBCluster) smartUpdate(sfs api.StatefulApp, cr *api
 
 	log.Info(fmt.Sprintf("primary pod is %s", primary))
 
-	waitLimit := 120
+	waitLimit := 2 * 60 * 60 // 2 hours
```

Comment:
I have a concern, If during the update we got `CrashLoopBackOff` we will be inside valid state for 2 hours.

Reply:
Will it help if we add 
```
			if pod.Status.Phase == corev1.PodFailed {
				return errors.Errorf("pod %s is in failed phase", pod.Name)
			}
```
when waiting for pod restart?

Reply:
yes, please

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/576/commits/5d105ade00a867151300f2ec3974df96183053a8